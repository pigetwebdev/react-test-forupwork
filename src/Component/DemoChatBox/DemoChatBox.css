.DemoChatBoxWrapper {
    position: relative;
    float: right;
    width: 600px;
    height: 100%;
}
.btn-mychat {    
    display: block;
    float: right;
    padding: 0;
    width: 60px;
    height: 60px;
    border-width: 0;
    background-image: url("../../asset/chat-bubble.png");
    background-size: contain;
    background-repeat: no-repeat;
    background-color: transparent;
    font-size: 0;
}
.btn-choiceselect1 {
    padding: 7px 12px;
    border-width: 0;
    margin-right: 7px;
    border-radius: 20px;
    background-color: #282c34;
    color: #fff;
}
.btn-choiceselect2 {
    padding: 5px 15px;
    border-width: 1;
    margin-right: 7px;
    border-color: #fff;
    border-radius: 20px;
    background-color:transparent;
    color: #fff;
}

.chat-show {
    float: left;
    margin-top: 0px;
    margin-bottom: 20px;
    padding: 11px 0;
    width: calc(100% - 60px);
    height: 50px;
    border-radius: 20px 0 0 20px;
    background-color: #fff;
}
.chat-nextshow {
    float: left;
    margin-top: 39px;    
    margin-bottom: 20px;
    padding: 11px 0;
    width: 100%;
    border-radius: 10px 10px 0 10px;
    background-color: #fd7e14;
    color: white
}
.first {
    position: absolute;
    top: 50%;
    right: 0;
    /* transform: translateY(-50%); */
    width: 100%;    
}
.second {
    position: absolute;
    top: 5%;
    right: 0;
    /* transform: translateY(-50%); */
    width: 100%;    
    
}
.btn-close {
    border-radius: 0;
    background-color: transparent;
    border-width: 0;
    color: #fff;
    position: absolute;
    top: 0;
    right: 0;    
}
