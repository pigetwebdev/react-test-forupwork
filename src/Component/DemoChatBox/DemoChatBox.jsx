import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { moveNextPage,moveLastPage } from '../../Actions/actionCreator';
import {bindActionCreators} from 'redux';
import './DemoChatBox.css';
const API = "http://localhost:3000/"     //API Base URL
const  DEFAULT_QUERY = 'resource/chatbottext.json'    //API

//import { Test } from './DemoChatBox.styles';

class DemoChatBox extends PureComponent { 
  constructor(props) {
    super(props);
    this.state = {
      chatBotLevel: 0,
      chatBotText : {},
      isLoading: false,
      finishState : false,
      error : null
    };
  }
  fetchChatBotText = () => {
    fetch(API + DEFAULT_QUERY)        
      .then(response => {console.log(response); return response.json()})
      .then(data =>
        {
          this.setState({
          chatBotText: data,
          isLoading: false,
        })
        }
      )
      .catch(error => this.setState({ error, isLoading: false }));
  }
  componentDidMount() {
    this.fetchChatBotText();
  }

  getButtons = (t=1) => {
    const {chatBotLevel,chatBotText,} = this.state;
    if(chatBotText.options && chatBotLevel)
    {
      return (<div>
        {
          chatBotText.options.map(option =>(
            <button key = {option.answer} type="button" onClick={() =>{ this.onClickAnswer(option.answer);} } className={"btn-choiceselect"+t}>{option.answer}</button>
          ))
        }
        <button type="button" onClick={ () => this.setState({ chatBotLevel: chatBotLevel+1}) } className={"btn-choiceselect"+t}>Comment</button>    
      </div>)
    }else
      return null;
  }
  onClickAnswer = (s) => {
    const {chatBotLevel,chatBotText,} = this.state;
    const node = chatBotText.options.find(option => (option.answer === s));
    this.setState({chatBotLevel: chatBotLevel+1, chatBotText: node },() => {console.log("Updated");console.log(chatBotText)});
    if(node.options === null) setTimeout(this.props.moveNextPage,3000);
  }
  render () {
    const {chatBotLevel,chatBotText,isLoading} = this.state;
    return (
      <div className="DemoChatBoxWrapper">
        {(chatBotLevel >1)&&
        <button className="btn-close" onClick = {this.props.moveLastPage}>Ã—</button>}
          {
            (chatBotLevel < 2) ?
            <div className="first">
              {chatBotLevel? <div className="chat-show" >{!isLoading? chatBotText.question: ""}  </div> : null }
              {(chatBotLevel < 2)&&<button type="button" onClick={ () => this.setState({ chatBotLevel: chatBotLevel===0?1:0}) } className="btn-mychat">Chat</button>}
              {this.getButtons()}
            </div>
          :<div className="second">
            {chatBotLevel? <div className="chat-nextshow" >{!isLoading? chatBotText.question: ""}  </div> : null }
            {this.getButtons(2)}
          </div>

          }

      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
      moveNextPage,
      moveLastPage
  }, dispatch)
}

export default connect(
  null,
  mapDispatchToProps,
)(DemoChatBox);
