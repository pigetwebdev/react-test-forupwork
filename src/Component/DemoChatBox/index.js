import DemoChatBox from './DemoChatBox';

export default DemoChatBox;
