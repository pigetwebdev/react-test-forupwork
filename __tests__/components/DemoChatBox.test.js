import React from 'react';
import DemoChatBox from '../../src/Component/DemoChatBox';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

const storeFake = state => {
  return {
    default: jest.fn(),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
    getState: () => state,
  };
};

test('DemoChatBox_____------______', () => {

    const store = storeFake({});  
    
    const component = renderer.create(
      <Provider store={store}>
      <DemoChatBox />
      </Provider>,
    );
        
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    // // manually trigger the callback
    //  tree.props.moveLastPage();
    // // // re-rendering
    //  tree = component.toJSON();
    //  expect(tree).toMatchSnapshot();
  
    // // manually trigger the callback
    // tree.props.onMouseLeave();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });