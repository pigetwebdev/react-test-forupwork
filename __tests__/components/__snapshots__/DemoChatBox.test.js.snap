// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoChatBox_____------______ 1`] = `
<div
  className="DemoChatBoxWrapper"
>
  <div
    className="first"
  >
    <button
      className="btn-mychat"
      onClick={[Function]}
      type="button"
    >
      Chat
    </button>
  </div>
</div>
`;
